src/openssl/lib_crypto.cr

```diff
@@ -1,5 +1,15 @@
-@[Link("crypto")]
+ifdef darwin
+  @[Link("crypto")]
+else
+  @[Link(ldflags: "`pkg-config --libs libcrypto`")]
+end
 lib LibCrypto
+  ifdef darwin
+    OPENSSL_102 = false
+  else
+    OPENSSL_102 = {{`(pkg-config --atleast-version=1.0.2 libcrypto && echo true) || echo false`.id}}
```

Comment:
Hence my question below as to whether we could detect that with `pkg-config`on OS X too.